### PR TITLE
Law & regs tooling - mobile

### PR DESCRIPTION
**Pull request by kewitham**
This PR adds the mobile functionality for the law and regs tooling:
Image

Image

Image

### CODE REVIEW

File: assets/js/laws-regs-parser.js [hunk 1]
```diff
@@ -56,16 +56,69 @@ export default function parseLawsAndRegs(mainContent) {
...
+  const mobileSearchBtn = document.querySelector('#mobile-search-button');
+  overlay.classList.add('display-none');
+  searchBoxWrapper.classList.remove('visible');
+  searchBoxWrapper.classList.remove('active');
+  searchInput.value = '';
+  mobileSearchBtn.classList.remove('display-none');
+  const sections = document.querySelectorAll('.section');
+  sections.forEach((section) => {
+    removeHighlights(section);
+    section.classList.remove('searched');
```
[the remark this answers is not in the document]
Comment (kewitham) [reply assets/js/laws-regs-parser.js]: Thanks for this! Made these changes - could you take a look and lmk what you think?

File: assets/js/laws-regs-parser.js [hunk 2]
```diff
@@ -258,39 +328,54 @@ function addPrintEventListeners(btn, divType) {
...
   const searchQuery = searchBox.value.toLowerCase();
   if (!searchQuery.length) {
     clearSearch(searchGo, searchNav, searchBox);
     return;
   }
   const sections = document.querySelectorAll('.section');
   sections.forEach((section) => {
-    removeHighlights(section);
-    updateSection(section, searchQuery);
+    let newSection = section.cloneNode(true);
```
Comment (okkays) [inline assets/js/laws-regs-parser.js:375]:
```suggestion
    const newSection = section.cloneNode(true);
```

File: assets/js/laws-regs-parser.js [hunk 3]
```diff
@@ -258,39 +328,54 @@ function addPrintEventListeners(btn, divType) {
...
+    newSection = removeHighlights(newSection);
+    newSection = updateSection(newSection, searchQuery);
+    section.parentNode.replaceChildren(newSection);
```
Comment (okkays) [inline assets/js/laws-regs-parser.js:378]: 👍 👍 nice

File: assets/js/laws-regs-parser.js [hunk 4]
```diff
@@ -258,39 +328,54 @@ function addPrintEventListeners(btn, divType) {
...
   location.hash = '#inPageResult' + newCount;
   currentCount.innerText = newCount;
 }
 
-function search() {
-  const searchBoxWrapper = document.querySelector('.search-box-wrapper');
+function search(searchBoxWrapper) {
   const searchBox = searchBoxWrapper.querySelector('.searchbox');
-  searchBox.focus();
+  // searchBox.focus();
```
Comment (okkays) [inline assets/js/laws-regs-parser.js:354]: (remove commented out code)

File: assets/js/laws-regs-parser.js [hunk 5]
```diff
@@ -44,28 +44,98 @@ export default function parseLawsAndRegs(mainContent) {
...
+}
+
+function initSearch(searchInput, searchBoxWrapper, searchGo, searchNav) {
   searchGo.classList.remove('display-none');
   searchNav.classList.add('display-none');
-  searchGo.addEventListener('click', search);
+  searchInput.addEventListener('keydown', (e) => {
+    if (e.key == 'Enter') {
+      search(searchBoxWrapper);
+    }
```
Comment (okkays) [inline assets/js/laws-regs-parser.js:117]: Jumping between results is now quick, but I'm still seeing slow performance on this event. The chrome profiler seems to indicate that it's calling this code ten times:

image

Ah - it looks like you're calling initSearch on input, which is going to add a keydown listener whenever you type into the box:

image

Yep - for this search, I typed `waffles the best I love them` then replaced it with `dog` and searched. The function is called once for every character in my search query:

image

**Review by okkays — CHANGES_REQUESTED** [review]
Nice! jumping between results is good now

Pressing enter still has a bug, but I think it's minor

File: assets/js/laws-regs-parser.js [hunk 6]
```diff
@@ -81,14 +83,16 @@ function setUpMobileSearch() {
   const searchBox = searchBoxWrapper.querySelector('.searchbox');
   setUpButtons(searchGo, searchNav, searchBox);
   searchInput.addEventListener('keydown', (e) => {
-    openOverlay(searchBoxWrapper, searchInput);
+    if (e.keyCode != 9) {
```
Comment (okkays) [inline assets/js/laws-regs-parser.js:86]: [keyCode is deprecated](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode)

it really looks like `key == 'Tab'` is the best alternative, as the replacement `code` [has terrible support](https://caniuse.com/?search=keyboardevent.code)

File: assets/js/laws-regs-parser.js [hunk 7]
```diff
@@ -108,15 +112,26 @@ function openOverlay(searchBoxWrapper, searchInput) {
   });
 }
 
+function listenForInput(e, searchBoxWrapper) {
+  console.log('hit input');
```
Comment (okkays) [inline assets/js/laws-regs-parser.js:116]: remove console log

File: assets/js/laws-regs-parser.js [hunk 8]
```diff
@@ -350,8 +365,8 @@ function navResults(e, dir, prevBtn, nextBtn, searchNav) {
 }
 
 function search(searchBoxWrapper) {
+  console.log('hit search');
```
Comment (okkays) [inline assets/js/laws-regs-parser.js:368]: nit: remove

File: assets/js/laws-regs-parser.js [hunk 9]
```diff
@@ -459,12 +483,16 @@ function clearSearch(searchGo, searchNav, searchBox, e = null) {
   }
   const sections = document.querySelectorAll('.section');
   searchGo.classList.add('display-none');
+  // searchGo.classList.remove('listening');
```
Comment (okkays) [inline assets/js/laws-regs-parser.js:486]: nit: remove

File: assets/js/laws-regs-parser.js [hunk 10]
```diff
@@ -68,9 +68,11 @@ export default function parseLawsAndRegs(mainContent) {
   const searchNav = searchBoxWrapper.querySelector('.result-nav');
   const searchBox = searchBoxWrapper.querySelector('.searchbox');
   setUpButtons(searchGo, searchNav, searchBox);
-  searchInput.addEventListener('input', () =>
-    initSearch(searchInput, searchBoxWrapper, searchGo, searchNav)
-  );
+  searchInput.addEventListener('input', () => {
+    // if (!searchGo.classList.contains('display-none')) {
```
Comment (okkays) [inline assets/js/laws-regs-parser.js:72]: remove commented out code

**Review by okkays — APPROVED** [review]
A bunch of small things; otherwise lgtm!

**Review by vwilliamsn — APPROVED** [review]
LGTM, great job @kewitham!

**Pull request closed without merging**